app/src/main/java/com/muzima/view/patients/PatientSummaryActivity.java

```diff
@@ -310,30 +529,55 @@ private String getObsByPatientUuidAndConceptId(String patientUuid, int conceptId
             observations = observationController.getObservationsByPatientuuidAndConceptId(patientUuid, conceptId);
             Concept concept = conceptController.getConceptById(conceptId);
             Collections.sort(observations, observationDateTimeComparator);
-            if(observations.size()>0){
+            if (observations.size() > 0) {
                 Observation obs = observations.get(0);
-                if(concept.isDatetime())
-                    return DateUtils.getFormattedDate(obs.getValueDatetime(),SIMPLE_DAY_MONTH_YEAR_DATE_FORMAT);
-                else if(concept.isCoded())
-                    return getConceptNameFromConceptNamesByLocale(obs.getValueCoded().getConceptNames(),applicationLanguage);
-                else if(concept.isNumeric())
+                if (concept.isDatetime())
+                    return DateUtils.getFormattedDate(obs.getValueDatetime(), SIMPLE_DAY_MONTH_YEAR_DATE_FORMAT);
+                else if (concept.isCoded())
+                    return getConceptNameFromConceptNamesByLocale(obs.getValueCoded().getConceptNames(), applicationLanguage);
+                else if (concept.isNumeric())
                     return String.valueOf(obs.getValueNumeric());
                 else
                     return obs.getValueText();
             }
-        } catch (ObservationController.LoadObservationException | Exception | ConceptController.ConceptFetchException e) {
+        } catch (ObservationController.LoadObservationException | Exception |
+                 ConceptController.ConceptFetchException e) {
             Log.e(getClass().getSimpleName(), "Exception occurred while loading observations", e);
         }
         return StringUtils.EMPTY;
     }
 
+    private Observation getObservationByPatientUuidAndConceptId(String patientUuid, int conceptId) throws JSONException, ObservationController.LoadObservationException {
+        List<Observation> observations = new ArrayList<>();
+        try {
+            observations = observationController.getObservationsByPatientuuidAndConceptId(patientUuid, conceptId);
+            Concept concept = conceptController.getConceptById(conceptId);
```

Comment:
the concept is not used in the function, should be removed

Reply:
Already solved on the last commit and instead of picking up the concept from conceptController... Picking it up from the obs object.